nbgitpuller/__init__.py

```diff
@@ -26,7 +46,22 @@ def load_jupyter_server_extension(nbapp):
             {'path': os.path.join(os.path.dirname(__file__), 'static')}
         )
     ]
-    web_app.settings['nbapp'] = nbapp
+    web_app.settings['app'] = app
```

Comment:
Ah, so this changes web_app.settings *globally* for all installed plugins, which I think is a mistake originally. I think we should just not touch web_app.settings at all, and use the `initialize` method in the tornado handler instead. The third item in the tuple passed to `add_handlers` can contain arbitrary dict that is passed as kwargs to an `initialize` method if found. Let's use that and stop putting things in the global settings?

Reply:
Ah! I went for adding a # FIXME note and force pushed to avoid making a reverting commit that can cause an unessecary merge conflict for Sean's plugin PR.

I opened #248 to represent this!